ui/lib/core/addon/components/text-file.js

```diff
@@ -0,0 +1,68 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+/**
+ * @module TextFile
+ * `TextFile` components render a file upload input with the option to toggle a "Enter as text" button
+ *  that changes the input into a textarea
+ *
+ * @example
+ * <TextFile
+ *  @hideTextAreaToggle={{true}}
+ *  @helpText="help text"
+ *  @onChange={{this.handleChange}}
+ *  @label="PEM Bundle"
+ * />
+ *
+ * @param {function} onChange - Callback function to call when the value of the input changes, returns an object in the shape of { value: fileContents, filename: 'some-file.txt' }
+ * @param {bool} [hideTextAreaToggle=false] - When true, renders a static file upload input and removes the option to toggle and input plain text
+ * @param {string} [helpText] - Text underneath label.
+ * @param {string} [label='File']  - Text to use as the label for the file input. If none, default of 'File' is rendered
+ */
+
+export default class TextFileComponent extends Component {
+  @tracked content = '';
+  @tracked filename = '';
+  @tracked uploadError = '';
+  @tracked showValue = false;
+  @tracked showTextArea = false;
+
+  async readFile(file) {
+    try {
+      this.content = await file.text();
+      this.filename = file.name;
+      this.handleChange();
+    } catch (error) {
+      this.clearFile();
+      this.uploadError = 'There was a problem uploading. Please try again.';
+    }
+  }
+
+  @action
+  handleFileUpload(e) {
+    e.preventDefault();
+    const { files } = e.target;
+    if (!files.length) return;
+    this.readFile(files[0]);
```

Comment:
removed iterating over `files` here because I didn't see a case for multiple files being uploaded (pgp-file does do this) - it also seemed like this was the only method that iterated as the file `index` wasn't returned to the parent